File: app/src/main/java/com/cxh/mvpart/ui/user/UserFragment.java
package com.cxh.mvpart.ui.user;

import android.widget.ImageView;
import android.widget.TextView;

import com.cxh.mvpart.R;
import com.cxh.mvpart.base.BaseFragment;
import com.cxh.mvpart.di.ActivityScoped;
import com.cxh.mvpart.util.GlideUtils;

import javax.inject.Inject;

import butterknife.BindView;

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2017/6/12
 */
@ActivityScoped
public class UserFragment extends BaseFragment implements UserContract.View {

    @BindView(R.id.startsTextView)
    TextView startsTextView;
    @BindView(R.id.showImage)
    ImageView showImage;

    @Inject
    UserPresenter mUserPresenter;

    @Inject
    UserFragment() {
    }

    @Override
    public int getLayoutID() {
        return R.layout.fragment_user;
    }

    @Override
    protected void initViewsAndEvents() {
        String path = "https://avatars0.githubusercontent.com/u/13111493?v=4&s=460";
        GlideUtils.loadImage(path, showImage);
    }

    @Override
    protected void refreshState() {
        mUserPresenter.takeView(this);
    }

    @Override
    public void onResume() {
        super.onResume();
        mUserPresenter.takeView(this);
    }

    @Override
    public void onPause() {
        mUserPresenter.dropView();
        super.onPause();
    }

    @Override
    public void setData(Object data) {
        startsTextView.setText((String) data);
    }
}

File: app/src/main/java/com/cxh/mvpart/ui/user/UserFragment.kt
package com.cxh.mvpart.ui.user

import com.cxh.mvpart.R
import com.cxh.mvpart.base.BaseFragment
import com.cxh.mvpart.di.ActivityScoped
import com.cxh.mvpart.util.GlideUtils
import kotlinx.android.synthetic.main.fragment_user.*
import javax.inject.Inject

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2017/6/12
 */
@ActivityScoped
class UserFragment @Inject constructor() : BaseFragment(), UserContract.View {

    @Inject lateinit var mUserPresenter: UserPresenter

    public override fun getLayoutID(): Int {
        return R.layout.fragment_user
    }

    override fun initViewsAndEvents() {
        val path = "https://avatars0.githubusercontent.com/u/13111493?v=4&s=460"
        GlideUtils.loadImage(path, showImage)
    }

    override fun refreshState() {
        mUserPresenter.takeView(this)
    }

    override fun onResume() {
        super.onResume()
        mUserPresenter.takeView(this)
    }

    override fun onPause() {
        mUserPresenter.dropView()
        super.onPause()
    }

    override fun setData(data: Any) {
        startsTextView.text = data as String
    }
}
